### PR TITLE
[EDA] Affiliation Settings Mappings Editing

### PR DESCRIPTION
**Pull request by MatthewBlanski**
# Critical Changes

# Changes

# Issues Closed

# New Metadata

## Unpackaged Metadata

# Deleted Metadata

# Testing Notes

### CODE REVIEW

File: force-app/main/default/classes/AffiliationMappingsService.cls [hunk 1]
```diff
@@ -56,20 +57,45 @@ public virtual with sharing class AffiliationMappingsService {
...
     /*****************************************************************************
-    * @description Retrieves an instance of the AffiliationMappingsMapper.
-    * @return The instance of AffiliationMappingsMapper.
-    ******************************************************************************/
+     * @description Checks user permissions, creates a model and passes to
+     * updateAffiliationMappings.
+     *****************************************************************************/
+    public void updateAffiliationMappings(String mappingName, String accRecordType, String conPrimaryAfflField) {
+        Boolean hasPermission = this.locateUserPermissionService().checkViewSetupForCurrentUser();
+        if (hasPermission == true) {
```
Comment (MatthewBlanski) [inline force-app/main/default/classes/AffiliationMappingsService.cls:74]: Why not make this a guard clause instead? If there is an exception thrown, then simply the check without the variable assignment could work as well.

File: force-app/main/default/lwc/primaryAffiliationsModalBody/primaryAffiliationsModalBody.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<template>
+    <div class="slds-grid slds-grid--vertical slds-grid_vertical-align-start slds-wrap">
+        <template if:true={modifyRecords}>
+            <div class="slds-col slds-size-1-of-1">{labelReference.modalBodyEditSave}</div>
+            <template if:true={accountRecordTypeComboboxVModel}>
+                <div class="slds-col slds-size-1-of-1">
```
Comment (MatthewBlanski) [inline force-app/main/default/lwc/primaryAffiliationsModalBody/primaryAffiliationsModalBody.html:6]: Let's try and make these comboboxes a little wider to give space for their dropdowns.

File: force-app/main/default/lwc/primaryAffiliationsModalBody/primaryAffiliationsModalBody.html [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<template>
+    <div class="slds-grid slds-grid--vertical slds-grid_vertical-align-start slds-wrap">
```
Comment (MatthewBlanski) [inline force-app/main/default/lwc/primaryAffiliationsModalBody/primaryAffiliationsModalBody.html:2]: This section likely needs a little padding so the combobox dropdowns resolve and the text is not quite so squished together. Look at the mock for sizing targets and the lightning design system modal documentation for how to get modal sizing. We may want a medium or large in this case.

**Review by MatthewBlanski — COMMENTED** [review]
We could use a number of changes on this item to address some styling difficulties.

**Comment by MatthewBlanski** [comment]
The changes look good and we've minimized scrolling. Approving this PR.

**Review by skosgi1 — APPROVED** [review]
Approved!

File: force-app/main/default/lwc/affiliationSettings/affiliationSettings.js [hunk 4]
```diff
@@ -6,6 +6,7 @@ import { ShowToastEvent } from "lightning/platformShowToastEvent";
 import getAffiliationsSettingsVModel from "@salesforce/apex/AffiliationsSettingsController.getAffiliationsSettingsVModel";
 import getPrimaryAffiliationsSettingsVModel from "@salesforce/apex/AffiliationsSettingsController.getPrimaryAffiliationsSettingsVModel";
 import updateAffiliationMappings from "@salesforce/apex/AffiliationsSettingsController.updateAffiliationMappings";
+import getRecordTypeLabelFromDeveloperName from "@salesforce/apex/UTIL_Describe.getRecordTypeLabelFromDeveloperName";
```
Comment (MatthewBlanski) [inline force-app/main/default/lwc/affiliationSettings/affiliationSettings.js:9]: I would prefer we not expose UTIL describe any further. If you need a success message, then please have the controller make a call to the record type service to get this information from a model.

File: force-app/main/default/lwc/affiliationSettings/affiliationSettings.js [hunk 5]
```diff
@@ -162,30 +161,19 @@ export default class affiliationSettings extends LightningElement {
...
+            conPrimaryAfflField: contactField,
         })
             .then((result) => {
                 this.showToast("success", "Save Complete", this.labelReference.successMessage.replace("{0}", result));
             })
+
             .catch((error) => {
                 // console.log('Inside error');
             });
+        this.refreshAllApex();
```
Comment (skosgi1) [inline force-app/main/default/lwc/affiliationSettings/affiliationSettings.js:176]: @MatthewBlanski wanted to check if we need refreshAllApex() here?

File: force-app/main/default/classes/AffiliationsSettingsController.cls [hunk 6]
```diff
@@ -47,6 +47,77 @@ public with sharing class AffiliationsSettingsController {
...
+    public static String updateAffiliationMappings(String mappingName, String accRecordType, String conPrimaryAfflField) {
+        try {
+            AffiliationsSettingsController.locateAffiliationMappingsService()
+                .updateAffiliationMappings(mappingName, accRecordType, conPrimaryAfflField);
+
+            RecordTypeModel accRecTypeModel = AffiliationsSettingsController.locateRecordTypeService()
+                .getRecordTypeModelFromRecTypeDevName('Account', accRecordType);
+            
+            if (accRecTypeModel == null) {
+                return '';
```
Comment (MatthewBlanski) [inline force-app/main/default/classes/AffiliationsSettingsController.cls:67]: Please return accRecordType here if it is null as invalid values can still be 'saved' if they are unchanged.

File: force-app/main/default/classes/AffiliationsSettingsController.cls [hunk 7]
```diff
@@ -47,6 +47,77 @@ public with sharing class AffiliationsSettingsController {
...
+     * @param conPrimaryAfflField
+     * @return An AffiliationsSettingsVModel
+     ************************************************************************************/
+    @AuraEnabled
+    public static String updateAffiliationMappings(String mappingName, String accRecordType, String conPrimaryAfflField) {
+        try {
+            AffiliationsSettingsController.locateAffiliationMappingsService()
+                .updateAffiliationMappings(mappingName, accRecordType, conPrimaryAfflField);
+
+            RecordTypeModel accRecTypeModel = AffiliationsSettingsController.locateRecordTypeService()
```
Comment (MatthewBlanski) [inline force-app/main/default/classes/AffiliationsSettingsController.cls:63]: Good use of a service here.

**Review by MatthewBlanski — COMMENTED** [review]
One tiny change requested.

**Review by MatthewBlanski — COMMENTED** [review]
Approved!

**Review by skosgi1 — APPROVED** [review]
Approved updates.

**Comment by MatthewBlanski** [comment]
Approved again!

**Pull request merged**